src/Functions/date_trunc.cpp

```diff
@@ -99,21 +114,24 @@ class FunctionDateTrunc : public IFunction
         }
         else
         {
-            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+            throw Exception(
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
                 "Number of arguments for function {} doesn't match: passed {}, should be 2 or 3",
-                getName(), arguments.size());
+                getName(),
+                arguments.size());
         }
 
         if (result_type_is_date)
-            return std::make_shared<DataTypeDate>();
+            return std::make_shared<DataTypeDateTime>();
         else
             return std::make_shared<DataTypeDateTime>(extractTimeZoneNameFromFunctionArguments(arguments, 2, 1));
```

Comment:
Here we always return the second option because we allowed a timezone argument for everything.